vars/byoScylladb.groovy

```diff
@@ -1,5 +1,22 @@
 #!groovy
 
+def list_as_comma_separated_str(input) {
+    // NOTE: transform all possible single-/multi region values into the scylla-pkg compatible format:
```

Comment:
isn't easier to read it as json, and then format it with comma separated string ?

Reply:
It supports 4 structures:
- list
- string with pickled list
- comma separated regions
- space separated regions

Reply:
but why we need to support all those variants that we aren't actually using

Reply:
i.e. SCT doesn't support space/comma separated for regions

Reply:
> but why we need to support all those variants that we aren't actually using

```
$ git grep "region_name" jenkins-pipelines/
...
test-cases/features/add-new-dc.yaml:region_name: 'eu-west-1 eu-west-2'
test-cases/longevity/longevity-ndbench-12-nodes-multidc-4h.yaml:region_name: 'us-east-1 eu-west-1'
test-cases/longevity/longevity-ndbench-192-nodes-multidc-4h.yaml:region_name: 'us-east-1 eu-west-1'
...
test-cases/manager/manager-regression-multiDC-set-distro.yaml:region_name: 'us-east-1 us-west-2'
...
test-cases/scylla-operator/longevity-scylla-operator-multidc-12h.yaml:region_name: 'eu-north-1 eu-west-1'
test-cases/scylla-operator/longevity-scylla-operator-multidc-3h.yaml:region_name: 'eu-north-1 eu-west-1'

```

Reply:
as you can see all those examples are in the test cases yamls, 
none of them in the pipelines, i.e. it won't work in the pipelines

since we need to pass it to SCT via environment variables, and we don't support all those variation for reading from env variables.

we are reading back from test-cases into the pipeline, but most for calculation of the pipeline timeouts, I don't think we read back the region, it's working only base on pipeline definitions

Reply:
anyhow you test it, and pass space separated region into one of our pipe, and see if it works. I'm betting it's not gonna work.

Reply:
`vars/getJenkinsLabels.groovy` fails trying to do the same thing you proposed.
One of us added this limitation here: https://github.com/scylladb/scylla-cluster-tests/pull/1099/files
From my point of view it is unneeded restriction.
I really would like to use `foo, bar, quuz` approach like we do with AMI ids and other parameters.

Reply:
As you can see now it can't be used, and pipeline currently assumes one specific option regarding the region parameter

SCT configuration is using multiple conflicting options to pass in lists, we can raise an issue to attend to that.

Anyhow I would want to support only one option if possible per a specific configuration option, I don't groovy/python variation for all of those.

Reply:
I added it as you requested but needed to add try-except block to support single-dc case with just name as string.

Reply:
We can have an issue for supporting all the options for lists, but we'll need to do it across all of the relevant values, in both ends, the pipelines and SCT.

For example ditch the env variable (or keep them for development only), and generate an extra configuration yaml file in pipelines ?

Anyhow let's move the discussion into a separate issue.